app/controllers/commodities_controller.rb

```diff
@@ -100,7 +100,7 @@ def heading_id
   end
 
   def show_validity_periods
-    @validity_periods = ValidityPeriod.all(Commodity, params[:id])
+    @validity_periods = ValidityPeriod.all(goods_nomenclature_class, params[:id])
```

Comment:
just an idea, it would be nice if we could use it 

`Heading.validity_period(params[:id])`
and
`Comodity.validity_period(params[:id])`

... maybe a future refactor

Reply:
Yeah, lets do this munchkin in a future refactor. I suspect this is quite a change to the api being driven in a few other places